src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Dialog.kt

```diff
@@ -20,45 +20,154 @@ package io.github.yuutoproject.yuutobot.commands
 
 import io.github.yuutoproject.yuutobot.commands.base.AbstractCommand
 import io.github.yuutoproject.yuutobot.commands.base.CommandCategory
+import io.github.yuutoproject.yuutobot.utils.EMOJI_REGEX
+import io.github.yuutoproject.yuutobot.utils.NONASCII_REGEX
+import io.github.yuutoproject.yuutobot.utils.httpClient
 import java.io.IOException
 import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
+import net.dv8tion.jda.api.utils.data.DataObject
 import net.dv8tion.jda.internal.utils.IOUtil
 import okhttp3.*
 import okhttp3.Callback as OkHttp3Callback
 import okhttp3.MediaType.Companion.toMediaType
 import okhttp3.RequestBody.Companion.toRequestBody
+import org.slf4j.LoggerFactory
 
-// TODO: get arch to implement the actual command
-class Dialog : AbstractCommand("dialog", CommandCategory.INFO, "does something", "[bg] <char> <text>") {
+class Dialog : AbstractCommand("dialog", CommandCategory.INFO, "Generates an image of a character in Camp Buddy saying anything you want", "[bg] <char> <text>") {
+    private val logger = LoggerFactory.getLogger(this.javaClass)
+
+    private val backgrounds = listOf(
+        "bath",
+        "beach",
+        "cabin",
+        "camp",
+        "cave",
+        "forest",
+        "messhall"
+    )
+
+    private val characters = listOf(
+        "aiden",
+        "avan",
+        "chiaki",
+        "connor",
+        "eduard",
+        "felix",
+        "goro",
+        "hiro",
+        "hunter",
+        "jirou",
+        "keitaro",
+        "kieran",
+        "knox",
+        "lee",
+        "naoto",
+        "natsumi",
+        "seto",
+        "taiga",
+        "yoichi",
+        "yoshi",
+        "yuri",
+        "yuuto"
+    )
+
+    private val backgroundsString = "`${backgrounds.joinToString("`, `")}`"
+    private val charactersString = "`${characters.joinToString("`, `")}`"
+
+    private val json = "application/json; charset=utf-8".toMediaType()
+
+    // Using ExperimentalStdLib for .removeFirst() in Mutable Lists
+    @ExperimentalStdlibApi
     override fun run(args: MutableList<String>, event: GuildMessageReceivedEvent) {
-        val moveThisToAStorageLocation = OkHttpClient()
+        val now = System.currentTimeMillis()
+
+        if (args.count() < 2) {
+            event.channel.sendMessage("This command requires two arguments : `dialog [background] <character> <text>` ([] is optional)").queue()
+            return
+        }
+
+        var character = args.removeFirst().toLowerCase()
```

Comment:
Why don't you use `removeAt(0)`

Reply:
Sorry, i got lost in cool kotlin builtin functions